module15/viin_product/models/viin_order.py

```diff
@@ -65,4 +73,4 @@ def change_to_canceled(self):
             if order.is_allowed_state(order.state, 'canceled'):
                 return self.env.ref('viin_product.viin_order_cancel_action').read()[0]
             else:
-                raise UserError("Changing order status from %s to %s is not allowed." % (order.state, 'canceled'))
\ No newline at end of file
+                raise UserError("Changing order status from %s to %s is not allowed." % (order.state, 'canceled'))
```

Comment:
raise lỗi nên cho phép dịch thuật string 
vd: raise UserError(**_(**"Changing order status from %s to %s is not allowed." % (order.state, 'canceled')**)**)
@duyquyen96